SimPEG/Regularization.py

```diff
@@ -416,655 +577,1031 @@ def eval2Deriv(self, m, v=None):
+            mesh=mesh, **kwargs
+        )
+
+
+    @property
+    def W(self):
+        """
+        Weighting matrix that takes the second spatial derivative in the
+        specified orientation
+        """
+        vol = self.regmesh.vol
+        if self.cell_weights is not None:
+            vol *= self.cell_weights
+
+        W = (
+            Utils.sdiag(vol**0.5) *
+            getattr(
+                self.regmesh,
+                'faceDiff{orientation}'.format(
+                    orientation=self.orientation
+                )
+            ) *
+            getattr(
+                self.regmesh,
+                'cellDiff{orientation}'.format(
+                    orientation=self.orientation
+                )
+            )
+        )
+        return W
+
+
+class Smooth_xx(BaseSmooth2):
+    """
+    Second-order smoothness along x. Regularizes on the second spatial
+    derivative in the x-direction
+    """
+
+    _multiplier_pair = 'alpha_xx'
+
+    def __init__(self, mesh, **kwargs):
+        super(Smooth_xx, self).__init__(
+            mesh=mesh, orientation='x', **kwargs
+        )
 
-            R(m) = \mathbf{W^\\top R^\\top R W (m-m_\\text{ref})}
 
-        The IRLS weights are recomputed after each beta solves.
-        It is strongly recommended to do a few Gauss-Newton iterations
-        before updating.
+class Smooth_yy(BaseSmooth2):
+    """
+    Second-order smoothness along y. Regularizes on the second spatial
+    derivative in the y-direction
     """
 
-    # set default values
-    eps_p = [1e-1, 1e-1, 1e-1]        # Threshold value for the model norm
-    eps_q = [1e-1, 1e-1, 1e-1]        # Threshold value for the model gradient norm
-    model = None     # Requires model to compute the weights
+    _multiplier_pair = 'alpha_yy'
 
-    l2model = None
-    gamma = 1.          # Model norm scaling to smooth out convergence
-    norms = [0., 2., 2., 2.] # Values for norm on (m, dmdx, dmdy, dmdz)
-    cell_weights = 1.        # Consider overwriting with sensitivity weights
-    nModels = 1 # Number of models
+    def __init__(self, mesh, **kwargs):
+        assert(mesh.dim > 1), (
+            "Mesh must have at least 2 dimensions to regularize along the "
+            "y-direction"
+        )
+        super(Smooth_yy, self).__init__(
+            mesh=mesh, orientation='y', **kwargs
+        )
 
-    def __init__(self, mesh, mapping=None, indActive=None, **kwargs):
-        Simple.__init__(self, mesh, mapping=mapping, indActive=indActive, **kwargs)
 
-        if isinstance(self.cell_weights,float):
-            self.cell_weights = np.ones(self.regmesh.nC) * self.cell_weights
+class Smooth_zz(BaseSmooth2):
+    """
+    Second-order smoothness along z. Regularizes on the second spatial
+    derivative in the z-direction
+    """
 
-        # if getattr(self, 'model', None) is None:
-        #             self.model = np.ones(self.regmesh.nC)
+    _multiplier_pair = 'alpha_zz'
```

Comment:
I think ditching these names and using `properties.link()` might be better? That way changes on each side will be reflected? Right now I think that it is down (i.e. from tik-->smooth) only.